tensorflow/python/data/kernel_tests/matching_files_dataset_op_test.py

```diff
@@ -0,0 +1,267 @@
+      next_element = itr.get_next()
+      sess.run(init_op)
+      with self.assertRaises(errors.OutOfRangeError):
+        sess.run(next_element)
+
+  def testSimpleDirectory(self):
+    filenames = ['a', 'b', 'c']
+    self._touchTempFiles(filenames)
+
+    dataset = MatchingFilesDataset(path.join(self.tmp_dir, '*'))
+    with self.cached_session() as sess:
+      itr = iterator_ops.Iterator.from_structure(dataset.output_types)
+      init_op = itr.make_initializer(dataset)
+      next_element = itr.get_next()
+      sess.run(init_op)
+
+      full_filenames = []
+      produced_filenames = []
+      for filename in filenames:
+        full_filenames.append(
+          compat.as_bytes(path.join(self.tmp_dir, filename)))
+        produced_filenames.append(compat.as_bytes(sess.run(next_element)))
+      self.assertItemsEqual(full_filenames, produced_filenames)
+      with self.assertRaises(errors.OutOfRangeError):
+        sess.run(itr.get_next())
+
+  def testSimpleDirectoryInitializer(self):
+    filenames = ['a', 'b', 'c']
+    self._touchTempFiles(filenames)
+
+    filename_placeholder = array_ops.placeholder(dtypes.string, shape=[])
+    dataset = MatchingFilesDataset(filename_placeholder)
+
+    with self.cached_session() as sess:
+      itr = iterator_ops.Iterator.from_structure(dataset.output_types)
+      init_op = itr.make_initializer(dataset)
+      next_element = itr.get_next()
+      sess.run(
+        init_op,
+        feed_dict={filename_placeholder: path.join(self.tmp_dir, '*')})
+
+      full_filenames = []
+      produced_filenames = []
+      for filename in filenames:
+        full_filenames.append(
+          compat.as_bytes(path.join(self.tmp_dir, filename)))
+        produced_filenames.append(compat.as_bytes(sess.run(next_element)))
+
+      self.assertItemsEqual(full_filenames, produced_filenames)
+
+      with self.assertRaises(errors.OutOfRangeError):
+        sess.run(itr.get_next())
+
+  def testFileSuffixes(self):
+    filenames = ['a.txt', 'b.py', 'c.py', 'd.pyc']
+    self._touchTempFiles(filenames)
+
+    filename_placeholder = array_ops.placeholder(dtypes.string, shape=[])
+    dataset = MatchingFilesDataset(filename_placeholder)
+
+    with self.cached_session() as sess:
+      itr = iterator_ops.Iterator.from_structure(dataset.output_types)
+      init_op = itr.make_initializer(dataset)
+      next_element = itr.get_next()
+      sess.run(
+        init_op,
+        feed_dict={filename_placeholder: path.join(self.tmp_dir, '*.py')})
+
+      full_filenames = []
+      produced_filenames = []
+      for filename in filenames[1:-1]:
+        full_filenames.append(
+          compat.as_bytes(path.join(self.tmp_dir, filename)))
+        produced_filenames.append(compat.as_bytes(sess.run(next_element)))
+      self.assertItemsEqual(full_filenames, produced_filenames)
+
+      with self.assertRaises(errors.OutOfRangeError):
+        sess.run(itr.get_next())
+
+  def testFileMiddles(self):
```

Comment:
Same comments here, passing in patterns as a constant will allow you to simplify the `next_element` op definition and replace "full" and "produced" with "expected" and "actual".